### ISSUE TITLE
Importing observations from a GIS system

### ISSUE THREAD

**Issue by mikkohei13** [luomus/laji#546]
_As someone having observations in a GIS system, I want to easily copy the data to Vihko without first converting it to a tabular format, so that the process is easy and quick for me._

 - Private people using Vihko
 - Organizations using Data bank, where secondary copy of data requires an unique identifier for each row

Possible solution: Allow importing Geopackage documents. Require the same data fields (i.e. attributes) as in tabular data imports, and allow the user to map the fields in similar way.

Open questions:

- What if the GIS file contains extra attributes, that are neither recognized nor needed by Vihko? Need to check how current UI (broken atm) handles these.

**Comment by mikkohei13** [comment]
@olzraiti @aorin or @wkmor1 Can you evaluate how large task this would be? (Should we have labels for estimated task size, in addition to "quick"?) Are there possible complications in handling Geopackage fields?

FYI @AlpoTurunen Testing mention feature here also.

**Comment by wkmor1** [comment]
I don't think this is a huge task. Just adding a step to convert the Geopackage file to something importable. Lots of tools to do this e.g., https://gdal.org/en/latest/programs/ogr2ogr.html. GIS software has identifiers. Do think there is any guarantee they are persistent though. Don't think any software can guarentee that. It is a question of how the users are using the identifiers. That situation is the same for any file format so has to be managed same way we always do.

**Comment by MLumme** [comment]
@mikkohei13 @AlpoTurunen  About the imported geometries, goeconverter returns them in a single field, but for import the system expects them to be a point in a split field, so before I start ripping everything apart to mangle them to right form, is the goal to support arbitrary geometries for the gis-import, or are we limiting them to a point too? I will need to do conversions of the WKT anyways to get coordinates out, so this is just for how exactly I need to modify the geometries and their validations.

**Comment by mikkohei13** [comment]
The import allows coordinates in two formats: A) split into lat, lon & system, and B) in a single field as WKT, ISO 6709, or GeoJSON. Latter should be used here.

**Comment by MLumme** [comment]
Alright, and seems my test file has something that causes the import to think there should be multiple fileds.

**Comment by rpulkka** [comment]
Conversion works, validation needs work.

**Comment by aorin** [comment]
Requires more planning (discussion on Slack)

**Comment by MLumme** [comment]
Testable here, https://546.dev.laji.fi/vihko/tools/import

**Comment by MLumme** [comment]
Removed from pending approval, needs some fixing with column mapping and gpkg-import does not work for tester.

**Comment by rpulkka** [comment]
New geoconverter can now convert the files, but Vihko data import can't do it yet. Prioritize this and assign to a developer.

**Comment by MLumme** [comment]
Updated with changes from newest dev branch and retested, original problem where tester could not import gpkg:s has gone away, probably some long since purged cache. Should now be good to go.
